jax/numpy/lax_numpy.py

```diff
@@ -2178,7 +2178,13 @@ def _can_call_numpy_array(x):
 
 # TODO(mattjj): maybe move these two functions into xla.py
 def _device_put_raw(x):
-  return array_result_handler(None, abstractify(x))(device_put(x))
+  try:
+    result = array_result_handler(None, abstractify(x))(device_put(x)) 
+  except:
```

Comment:
Can we catch some specific exception classes here, rather than the generic `except`?

Reply:
Sure! I have added `RuntimeError`, `TypeError` and `ValueError` which I thought would be relevant. Please let me know if you want me to add or remove any class.